spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/messaging/SleuthKafkaStreamsConfiguration.java

```diff
@@ -66,8 +67,8 @@ static KafkaStreamsTracing kafkaStreamsTracing(Tracing tracing) {
 
 	@Bean
 	static KafkaStreamsBuilderFactoryBeanPostProcessor kafkaStreamsBuilderFactoryBeanPostProcessor(
-			KafkaStreamsTracing kafkaStreamsTracing) {
-		return new KafkaStreamsBuilderFactoryBeanPostProcessor(kafkaStreamsTracing);
+			BeanFactory beanFactory) {
```

Comment:
Maybe `ObjectProvider<KafkaStreamsTracing>` so the type is not lost.

Reply:
Good suggestion, thanks! Applied just now.